provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/contract/amqp/ContractAPIAmqp.java

```diff
@@ -168,8 +169,10 @@ public synchronized void authenticateWithProvisioningService(RequestData request
             this.amqpSaslHandler.setSasToken(requestData.getSasToken());
         }
 
+        byte[] payload = new DeviceRegistrationParser(requestData.getRegistrationId(), customPayload).toJson().getBytes();
+
         // SRS_ContractAPIAmqp_07_005: [This method shall send an AMQP message with the property of iotdps-register.]
-        this.provisioningAmqpOperations.sendRegisterMessage(responseCallback, callbackContext);
+        this.provisioningAmqpOperations.sendRegisterMessage(responseCallback, callbackContext, payload);
     }
```

Comment:
Is it possible to send response payload in responseCallback ? Is addional param neeed?

Reply:
For this one I had to add the payload because this is the payload that gets sent to the server and there is not other object I can piggyback on.

---
In reply to: [279462361](https://github.com/Azure/azure-iot-sdk-java/pull/487#discussion_r279462361) [](ancestors = 279462361)

Reply:
If you see another way I'm will to move it, but I can't find it.

---
In reply to: [279582289](https://github.com/Azure/azure-iot-sdk-java/pull/487#discussion_r279582289) [](ancestors = 279582289,279462361)